src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -343,67 +347,72 @@ def deleteBlocks(self):
 
         blocksToDelete = []
         containerDict = {}
-        # Populate containerDict, assigning each block to its correspondant container
-        for blockName in blockDict:
-            container = blockDict[blockName]['dataset']
-            # If the container is not in the dictionary, create a new entry for it
-            if container not in containerDict:
-                # Set of sites to which the container needs to be transferred
-                sites = set(x.replace("_MSS", "_Tape") for x in blockDict[blockName]['sites'])
-                containerDict[container] = {'blocks': [], 'rse': sites}
-            containerDict[container]['blocks'].append(blockName)
-
-        for contName in containerDict:
-            cont = containerDict[contName]
-
-            # Checks if the container is not requested in any sites.
-            # This should never be triggered, but better safe than sorry
-            if not cont['rse']:
-                logging.warning("No rules for container: %s. Its blocks won't be deleted.", contName)
-                continue
+        for blocksSlice in grouper(blockDict, self.sliceSize):
+            logging.info("Handeling %d candidate blocks", len(blocksSlice))
+            # Populate containerDict, assigning each block to its correspondant container
+            for blockName in blocksSlice:
+                container = blockDict[blockName]['dataset']
+                # If the container is not in the dictionary, create a new entry for it
+                if container not in containerDict:
+                    # Set of sites to which the container needs to be transferred
+                    sites = set(x.replace("_MSS", "_Tape") for x in blockDict[blockName]['sites'])
+                    containerDict[container] = {'blocks': [], 'rse': sites}
+                containerDict[container]['blocks'].append(blockName)
+
+            for contName in containerDict:
+                cont = containerDict[contName]
+
+                # Checks if the container is not requested in any sites.
+                # This should never be triggered, but better safe than sorry
+                if not cont['rse']:
+                    logging.warning("No rules for container: %s. Its blocks won't be deleted.", contName)
+                    continue
 
-            try:
-                # Get RSE in which each block is available
-                availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'])
-            except Exception as exc:
-                msg = "Failed to get replica info for blocks in container: %s.\n" % contName
-                msg += "Will retry again in the next cycle. Error: %s" % str(exc)
-                logging.error(msg)
-                continue
+                try:
```

Comment:
I am actually concerned with all these logical changes here, especially because we don't have tests for the deletion logic. Can't you keep most of the existent logic and change only this try except to use the block slicing parameter? It would become something like:
```
availableRSEs = self.rucio.getReplicaInfoForBlocks(block=sliceOfBlocks, deep=self.deep)
```
and then you refer to this `sliceOfBlocks` in the code below, when required.

Reply:
The logic here hasn't changed. I'm only using the `sliceOfBlocks` to populate the containers dict. Everything else does exactly the same as before, but using the smaller container dictionary.